sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/ScalaReflectionSuite.scala

```diff
@@ -109,6 +109,64 @@ object TestingUDT {
   }
 }
 
+/** An example derived from Twitter/Scrooge codegen for thrift  */
+object ScroogeLikeExample {
+  def apply(
+    x: Int
+  ): ScroogeLikeExample =
+    new Immutable(
+      x
+    )
+
+  def unapply(_item: ScroogeLikeExample): _root_.scala.Option[Int] = _root_.scala.Some(_item.x)
+
+  class Immutable(val x: Int) extends ScroogeLikeExample
+
+  trait Proxy extends ScroogeLikeExample {
+    protected def _underlying_ScroogeLikeExample: ScroogeLikeExample
+    override def x: Int = _underlying_ScroogeLikeExample.x
+  }
+}
+
+trait ScroogeLikeExample
+  extends _root_.scala.Product1[Int]
+  with java.io.Serializable
+{
+  import ScroogeLikeExample._
+
+  def x: Int
+
+  def _1: Int = x
```

Comment:
override?

Reply:
sure, i can't figure out if it is strictly necessary though